python/lsst/ts/standardscripts/base_focus_sweep.py

```diff
@@ -0,0 +1,361 @@
+                type: string
+                enum: ["x", "y", "z", "u", "v"]
+              focus_window:
+                description: Total range (window) for the focus sweep.
+                type: number
+              n_steps:
+                description: Number of steps to take inside the focus window.
+                type: number
+                minimum: 3
+              n_images_per_step:
+                description: Number of images to take at each focus position.
+                type: integer
+                default: 1
+              program:
+                description: >-
+                    Optional name of the program this dataset belongs to.
+                type: string
+                default: FOCUS_SWEEP
+              reason:
+                description: Optional reason for taking the data.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              ignore:
+                description: >-
+                    CSCs from the group to ignore in status check. Name must
+                    match those in self.group.components, e.g.; hexapod_1.
+                type: array
+                items:
+                  type: string
+            required: ["axis", "focus_window", "n_steps"]
+            additionalProperties: false
+        """
+        schema_dict = yaml.safe_load(schema_yaml)
+
+        base_schema_dict = super().get_schema()
+
+        for properties in base_schema_dict["properties"]:
+            schema_dict["properties"][properties] = base_schema_dict["properties"][
+                properties
+            ]
+
+        return schema_dict
+
+    async def configure(self, config: types.SimpleNamespace) -> None:
+        """Configure script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+        await self.configure_tcs()
+        await self.configure_camera()
+        await self.configure_ocps()
+
+        if hasattr(config, "ignore"):
+            for comp in config.ignore:
+                if comp in self.tcs.components_attr:
+                    self.log.debug(f"Ignoring TCS component {comp}.")
+                    setattr(self.tcs.check, comp, False)
+                elif comp in self.camera.components_attr:
+                    self.log.debug(f"Ignoring Camera component {comp}.")
+                    setattr(self.camera.check, comp, False)
+                else:
+                    self.log.warning(
+                        f"Component {comp} not in CSC Groups. "
+                        f"Must be one of {self.tcs.components_attr} or "
+                        f"{self.camera.components_attr}. Ignoring."
+                    )
+
+        # Edge case checks
+        if config.n_steps < 3:
+            raise ValueError(
```

Comment:
If you really want to keep this check (which I think you don't need to), please, replace this with a `RuntimeError`.